### ISSUE TITLE
strerror_r on cygwin

### ISSUE THREAD

**Issue by kubo** [oracle/odpi#138]
ODPI-C version: 4.0.1
OS: [Cygwin](https://cygwin.com/) on Windows
Compiler: gcc

[Cygwin declares GNU-version strerror_r when `_GNU_SOURCE` is defined](https://cygwin.com/git/?p=newlib-cygwin.git;a=blob;f=newlib/libc/include/string.h;h=60e837bc08910493afbdffa509303cb642c1b892;hb=HEAD#l92). However the POSIX version is used in [the following code](https://github.com/oracle/odpi/blob/v4.0.1/src/dpiError.c#L259-L263).
```c
#if defined(__GLIBC__)
    message = strerror_r(err, buffer, sizeof(buffer));
#else
    message = (strerror_r(err, buffer, sizeof(buffer)) == 0) ? buffer : NULL;
#endif
```

No warnings by accident when compiling ODPI-C. That's because comparison with a pointer and literal zero is compilable without warnings.

When the above code is changed as follows,
```c
#if defined(__GLIBC__)
    message = strerror_r(err, buffer, sizeof(buffer));
#else
    int rv = strerror_r(err, buffer, sizeof(buffer));
    message = (rv == 0) ? buffer : NULL;
#endif
```
gcc prints the following warning.
```shell
gcc -c -Iinclude -O2 -g -Wall -Wextra -fPIC src/dpiError.c -o build/dpiError.o
src/dpiError.c: In function 'dpiError__setFromOS':
src/dpiError.c:262:14: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
     int rv = strerror_r(err, buffer, sizeof(buffer));
              ^~~~~~~~~~
```
This implies that the strerror_r isn't the POSIX version returning `int`. It is the GNU version returning `char *`.

The following code in place of `#if defined(__GLIBC__)`  fixes the issue.
```c
#if defined(__GLIBC__) || defined(__CYGWIN__)
```

**Comment by cjbj** [comment]
Thanks @kubo

**Comment by anthony-tuininga** [comment]
I've marked this for version 4.1 but if a patch release (4.0.2) is created I'll include it in there, too.

**Comment by anthony-tuininga** [comment]
Let me know if the creation of a patch release would be helpful to you, @kubo.

**Comment by anthony-tuininga** [comment]
This was included in 4.0.2 released on August 31, 2020.
